test/prompt/invocation_layer/test_cohere.py

```diff
@@ -54,106 +54,153 @@ def test_invoke_with_no_kwargs():
-    else:
-        layer = CohereInvocationLayer(model_name_or_path="command", api_key="fake_key")
+
+    layer = CohereInvocationLayer(model_name_or_path="command", api_key="fake_key", stream=True)
 
     with unittest.mock.patch("haystack.nodes.prompt.invocation_layer.CohereInvocationLayer._post") as mock_post:
-        # Mock the response, need to return a list of dicts
-        mock_post.return_value = MagicMock(text='{"generations":[{"text": "Hello"}]}')
+        # Mock the response
+        mock_post.return_value = Mock(iter_lines=Mock(return_value=['{"text": "Hello"}', '{"text": " there"}']))
+        layer.invoke(prompt="Tell me hello")
+
+    assert mock_post.called
+
+    # Check if stop_words are passed to _post as stop parameter
+    _, called_kwargs = mock_post.call_args
+
+    # stream is always passed to _post
+    assert "stream" in called_kwargs and called_kwargs["stream"]
+
+
+@pytest.mark.unit
+def test_streaming_stream_param_from_init_no_stream(mock_auto_tokenizer):
+    """
+    Test stream parameter is correctly passed from PromptNode to wire in CohereInvocationLayer from init
+    """
+
+    layer = CohereInvocationLayer(model_name_or_path="command", api_key="fake_key")
+
+    with unittest.mock.patch("haystack.nodes.prompt.invocation_layer.CohereInvocationLayer._post") as mock_post:
+        # Mock the response
+        mock_post.return_value = Mock(text='{"generations":[{"text": "Hello there"}]}')
+        layer.invoke(prompt="Tell me hello")
+
+    assert mock_post.called
+
+    # Check if stop_words are passed to _post as stop parameter
+    _, called_kwargs = mock_post.call_args
+
+    # stream is always passed to _post
+    assert "stream" in called_kwargs
+    assert not bool(called_kwargs["stream"])
+
+
+@pytest.mark.unit
+def test_streaming_stream_param_from_invoke(mock_auto_tokenizer):
+    """
+    Test stream parameter is correctly passed from PromptNode to wire in CohereInvocationLayer from invoke
+    """
+    layer = CohereInvocationLayer(model_name_or_path="command", api_key="fake_key")
+
+    with unittest.mock.patch("haystack.nodes.prompt.invocation_layer.CohereInvocationLayer._post") as mock_post:
+        # Mock the response
+        mock_post.return_value = Mock(iter_lines=Mock(return_value=['{"text": "Hello"}', '{"text": " there"}']))
+        layer.invoke(prompt="Tell me hello", stream=True)
+
+    assert mock_post.called
+
+    # Check if stop_words are passed to _post as stop parameter
+    _, called_kwargs = mock_post.call_args
+
+    # stream is always passed to _post
+    assert "stream" in called_kwargs
+    assert bool(called_kwargs["stream"])
 
-        if using_constructor:
-            layer.invoke(prompt="Tell me hello")
-        else:
-            layer.invoke(prompt="Tell me hello", stream=stream)
 
-        assert mock_post.called
+@pytest.mark.unit
+def test_streaming_stream_param_from_invoke_no_stream(mock_auto_tokenizer):
```

Comment:
I think we should adapt the test such that we inititalize `CohereInvocationLayer` with `stream=True` and call `layer.invoke` with `stream=False` to test that we can change the streaming behavior from invoke to no stream. Otherwise, this test case is identical with `test_streaming_stream_param_from_init_no_stream`.

Reply:
I agree, correcting!